internal/js/modules/k6/browser/common/element_handle.go

```diff
@@ -156,132 +156,12 @@ func (h *ElementHandle) click(p *Position, opts *MouseClickOptions) error {
-		return 0, 0, fmt.Errorf("getting inner width and height: %w", err)
-	}
-
-	m, ok := v.(map[string]any)
-	if !ok {
-		return 0, 0, fmt.Errorf("unexpected value %q when getting inner width and height", v)
-	}
-
-	width, ok := m["width"].(float64)
-	if !ok {
-		return 0, 0, fmt.Errorf("unexpected value %q when getting inner width", m["width"])
-	}
-
-	height, ok := m["height"].(float64)
-	if !ok {
-		return 0, 0, fmt.Errorf("unexpected value %q when getting inner height", m["height"])
-	}
-
-	return int64(width), int64(height), nil
-}
-
-func filterQuads(viewportWidth, viewportHeight int64, quads []dom.Quad) (*Position, error) {
-	var filteredQuads []dom.Quad
-	for _, q := range quads {
-		// Keep the points within the viewport and positive.
-		nq := q
-		for i := 0; i < len(q); i += 2 {
-			nq[i] = math.Min(math.Max(q[i], 0), float64(viewportWidth))
-			nq[i+1] = math.Min(math.Max(q[i+1], 0), float64(viewportHeight))
-		}
-		// Compute sum of all directed areas of adjacent triangles
-		// https://en.wikipedia.org/wiki/Polygon#Area
-		var area float64
-		for i := 0; i < len(q); i += 2 {
-			p2 := (i + 2) % (len(q) / 2)
-			area += (q[i]*q[p2+1] - q[p2]*q[i+1]) / 2
-		}
-		// We don't want to click on something less than a pixel.
-		if math.Abs(area) > 0.99 {
-			filteredQuads = append(filteredQuads, q)
-		}
-	}
-	if len(filteredQuads) == 0 {
-		return nil, errors.New("node is either not visible or not an HTMLElement")
-	}
-
-	// Return the middle point of the first quad.
-	var (
-		first = filteredQuads[0]
-		l     = len(first)
-		n     = (float64(l) / 2)
-		p     Position
-	)
-	for i := 0; i < l; i += 2 {
-		p.X += first[i] / n
-		p.Y += first[i+1] / n
-	}
-
-	p = compensateHalfIntegerRoundingError(p)
-
-	return &p, nil
-}
-
-// Firefox internally uses integer coordinates, so 8.5 is converted to 9 when clicking.
-//
-// This does not work nicely for small elements. For example, 1x1 square with corners
-// (8;8) and (9;9) is targeted when clicking at (8;8) but not when clicking at (9;9).
-// So, clicking at (8.5;8.5) will effectively click at (9;9) and miss the target.
-//
-// Therefore, we skew half-integer values from the interval (8.49, 8.51) towards
-// (8.47, 8.49) that is rounded towards 8. This means clicking at (8.5;8.5) will
-// be replaced with (8.48;8.48) and will effectively click at (8;8).
-//
-// Other browsers use float coordinates, so this change should not matter.
-func compensateHalfIntegerRoundingError(p Position) Position {
-	if rx := p.X - math.Floor(p.X); rx > 0.49 && rx < 0.51 {
-		p.X -= 0.02
-	}
-	if ry := p.Y - math.Floor(p.Y); ry > 0.49 && ry < 0.51 {
-		p.Y -= 0.02
-	}
-	return p
+// This will get the clickable point of the element relative to the page even
+// when the element is in an iframe. In some cases this isn't the case and we
+// need to translate the point to the page.
+func (h *ElementHandle) clickablePoint() *Position {
+	r := h.BoundingBox()
+	return &Position{X: r.X + r.Width/2, Y: r.Y + r.Height/2}
```

Comment:
I believe this significant change requires more testing, as it seems to alter behavior that affects every pointer action. Do you think we have enough tests to make this change?